dss/operations/params.py

```diff
@@ -0,0 +1,146 @@
+            yield name
+        except lambda_client.exceptions.ResourceNotFoundException:
+            print(f"{name} not deployed, or does not deploy a Lambda function")
+
+def get_elasticsearch_endpoint():
+    es_client = boto3.client("es")
+    domain_name = os.environ['DSS_ES_DOMAIN']
+    domain_info = es_client.describe_elasticsearch_domain(DomainName=domain_name)
+    return domain_info['DomainStatus']['Endpoint']
+
+def get_admin_user_emails():
+    secret_base = "{}/{}/".format(
+        os.environ['DSS_SECRETS_STORE'],
+        os.environ['DSS_DEPLOYMENT_STAGE'])
+
+    gcp_secret_id = secret_base + os.environ['GOOGLE_APPLICATION_CREDENTIALS_SECRETS_NAME']
+    admin_secret_id = secret_base + os.environ['ADMIN_USER_EMAILS_SECRETS_NAME']
+    resp = boto3.client("secretsmanager").get_secret_value(SecretId=gcp_secret_id)
+    gcp_service_account_email = json.loads(resp['SecretString'])['client_email']
+    resp = boto3.client("secretsmanager").get_secret_value(SecretId=admin_secret_id)
+    admin_user_emails = [email for email in resp['SecretString'].split(',') if email.strip()]
+    admin_user_emails.append(gcp_service_account_email)
+    return ",".join(admin_user_emails)
+
+params = dispatch.target(
+    "params",
+    arguments={},
+    help=__doc__
+)
+
+@params.action("ssm-print")
+def ssm_print(argv: typing.List[str], args: argparse.Namespace):
+    """Print out all environment variables stored in the SSM store"""
+    ssm_env = get_ssm_lambda_environment()
+    for name, val in ssm_env.items():
+        print(f"{name}={val}")
+
+@params.action("ssm-push")
+def ssm_push(argv: typing.List[str], args: argparse.Namespace):
+    """Push a new environment variable into the SSM store"""
+    name, val = args.split("=")
+
+    # Set the variable in the SSM store
+    ssm_env = get_ssm_lambda_environment()
+    ssm_env[name] = val
+    set_ssm_lambda_environment(ssm_env)
+
+@params.action("lambda-print")
+def lambda_print(argv: typing.List[str], args: argparse.Namespace):
+    """Print out the current environment of all deployed lambda functions"""
+    for lambda_name in get_deployed_lambdas():
+        lambda_env = get_deployed_lambda_environment(lambda_name)
+        print(f"\n{lambda_env}:")
+        for name, val in lambda_env.items():
+            print(f"{name}={val}")
+
+@params.action("lambda-set")
+def lambda_set(argv: typing.List[str], args: argparse.Namespace):
+    """Set a single environment variable into each deployed lambda"""
+    name, val = args.split("=")
+
+    # Set the variable in the SSM store first
+    ssm_env = get_ssm_lambda_environment()
+    ssm_env[name] = val
+    set_ssm_lambda_environment(ssm_env)
+
+    # Set the variable in each lambda function
+    for lambda_name in get_deployed_lambdas():
+        lambda_env = get_deployed_lambda_environment(lambda_name)
+        lambda_env[name] = val
+        set_deployed_lambda_environment(lambda_name, lambda_env)
+
+@params.action("lambda-unset")
```

Comment:
action arguments should be explicit, reduces "happy little accidents" 
see `dss/operations/elasticsearch.py line 93`

Reply:
To clarify - do you mean I should add a flag like `--lambda-name` for the user to specify the name of the lambda function, rather than using `argv` and `args`?

Reply:
Leaving this as-is. In the code we currently set the given environment variable for the param store, then for all lambda functions. Since we don't want each lambda function to have its own environment variable values, this is how we want to do it.